kratos/utilities/python_function_callback_utility.h

```diff
@@ -92,274 +88,13 @@ class PythonGenericFunctionUtility
-    bool mUseLocalSystem = false;                /// If we use a local system
-    BoundedMatrix<double, 3, 3> mRotationMatrix; /// The rotation matrix
-    array_1d<double, 3> mCenterCoordinates;      /// The center of coordinates
-
-    ///@}
 }; /// PythonGenericFunctionUtility
 
-/**
- * @class ApplyFunctionToNodesUtility
- * @ingroup KratosCore
- * @brief This function applies a givn function to its nodes calling PythonGenericFunctionUtility
- * @details The functions can be constructed by providing a python-defined method of the type
- * @author Riccardo Rossi
- */
-class ApplyFunctionToNodesUtility
-{
-public:
-    ///@name Type definitions
-    ///@{
-
-    /// The index type definition
-    typedef std::size_t IndexType;
-
-    /// Counted pointer of ApplyFunctionToNodesUtility
-    KRATOS_CLASS_POINTER_DEFINITION(ApplyFunctionToNodesUtility);
-
-    ///@}
-    ///@name Life Cycle
-    ///@{
-
-    /**
-     * @brief Default constructor
-     * @param rNodes The nodes where to set the values
-     * @param pFunction The function to set
-     */
-
-    ApplyFunctionToNodesUtility(
-        ModelPart::NodesContainerType& rNodes,
-        PythonGenericFunctionUtility::Pointer pFunction
-        ): mrNodes(rNodes),
-           mpFunction(pFunction)
-    {
-
-    };
-
-    ///@}
-    ///@name Operations
-    ///@{
-
-    /**
-     * @brief This method applies the function for a given variable at a certain time
-     * @param rVariable The variable type
-     * @param t The time variable
-     * @tparam TVarType The type of variable considered
-     */
-    template<class TVarType>
-    void ApplyFunction(
-        const TVarType& rVariable,
-        const double t
-        )
-    {
-        // The first node iterator
-        const auto it_node_begin = mrNodes.begin();
-
-        if(!mpFunction->UseLocalSystem()) {
-            //WARNING: do NOT put this loop in parallel, the python GIL does not allow you to do it!!
-            for (IndexType k = 0; k< mrNodes.size(); k++) {
-                auto it_node = it_node_begin + k;
-                const double value = mpFunction->CallFunction(it_node->X(), it_node->Y(), it_node->Z(), t, it_node->X0(), it_node->Y0(), it_node->Z0());
-                it_node->FastGetSolutionStepValue(rVariable) = value;
-            }
-        } else {
-            //WARNING: do NOT put this loop in parallel, the python GIL does not allow you to do it!!
-            for (IndexType k = 0; k< mrNodes.size(); k++) {
-                auto it_node = it_node_begin + k;
-                const double value = mpFunction->RotateAndCallFunction(it_node->X(), it_node->Y(), it_node->Z(), t, it_node->X0(), it_node->Y0(), it_node->Z0());
-                it_node->FastGetSolutionStepValue(rVariable) = value;
-            }
-        }
-    }
-
-    /**
-     * @brief This method returns all the evaluated values in a given time
-     * @param t The time variable
-     */
-    std::vector<double> ReturnFunction(const double t)
```

Comment:
@RiccardoRossi FYI this was your implementation, I just moved the code. The name was already like that

Reply:
well, from what i see just above there was a doxygen explaining it....now it is gone...

Reply:
It is now in the header. In the cpp there is no Oxygen